app/controllers/api/internal/v2/prosecution_case_laa_references_controller.rb

```diff
@@ -37,18 +35,13 @@ def enforce_contract!(contract)
           end
         end
 
-        def unlink
-          check_defendant_presence!
+        def unlink_defendant!
+          defendant_id = transformed_params[:defendant_id]
```

Comment:
Thinking about it, you could simply call `LaaReference.retrieve_by_defendant_id_and_optional_maat_reference(transformed_params[:defendant_id]` here, as that will now raise an `ActiveRecord::RecordNotFound` with an appropriate error message all by itself.

Reply:
good idea, something for later! ... I'm releasing it now :)